featuretools/entityset/serialization.py

```diff
@@ -1,75 +1,188 @@
-import logging
+# -*- coding: utf-8 -*-
+
+import json
 import os
 import shutil
+import sys
+import uuid
 from tempfile import mkdtemp
 
-from pandas import Timestamp
-from pandas.io.pickle import read_pickle as pd_read_pickle
-from pandas.io.pickle import to_pickle as pd_to_pickle
+import numpy as np
+import pandas as pd
+from pandas.io.pickle import (to_pickle as pd_to_pickle,
+                              read_pickle as pd_read_pickle)
 
-from featuretools import variable_types as vtypes
+import featuretools.variable_types.variable as vtypes
 
-logger = logging.getLogger('featuretools.entityset')
 
-_datetime_types = vtypes.PandasTypes._pandas_datetimes
+def read_parquet(path):
+    return read_entityset(path)
 
 
-def to_pickle(entityset, path):
-    """Save the entityset at the given path.
+def read_pickle(path):
+    return read_entityset(path)
 
-       Args:
-           entityset (:class:`featuretools.BaseEntitySet`) : EntitySet to save
-           path : pathname of a directory to save the entityset
-            (includes a CSV file for each entity, as well as a metadata
-            pickle file)
 
-    """
+def read_entityset(path):
+    from featuretools.entityset.entityset import EntitySet
+    entityset_path = os.path.abspath(os.path.expanduser(path))
+    with open(os.path.join(entityset_path, 'metadata.json')) as f:
+        metadata = json.load(f)
+    return EntitySet.from_metadata(metadata, root=entityset_path,
+                                   load_data=True)
+
+
+def load_entity_data(metadata, dummy=True, root=None):
```

Comment:
let's split this into two functions. one that does variable types and one that does data